ArduPlane/tiltrotor.cpp

```diff
@@ -310,24 +341,89 @@ void Tiltrotor::continuous_update(void)
+                            // use the VTOL flight tilt up rate if the VTOL flight tilt down rate isn't set
+                            tilt_rate_dps = max_rate_up_dps;
+                        }
+                    }
+                    new_tilt = get_forward_flight_tilt();
+                } else if (_transition_fwd_tilt_frac > deg2quad((float)max_angle_deg)) {
+                    // If the maximum throttle of the tilting motors is clipping and rotors are tilted past the normal
+                    // VTOL limit, tilt the rotors back faster to prevent possible loss of attitude control.
+                    if (max_rate_down_transition_dps > 0) {
+                        tilt_rate_dps = max_rate_down_transition_dps;
+                    } else if (max_rate_down_dps > 0) {
+                        // use the VTOL flight tilt down rate if the rate to be used when past Q_TILT_MAX isn't set
+                        tilt_rate_dps = max_rate_down_dps;
+                    } else {
+                        // use the VTOL flight tilt up rate if the VTOL flight tilt down rate isn't set
+                        tilt_rate_dps = max_rate_up_dps;
+                    }
+                    tilt_rate_dps = (3 * tilt_rate_dps) / 2;
+                    new_tilt = deg2quad((float)max_angle_deg);
+
+                }
+
+                AP::logger().Write("QTLT", "TimeUS,MT,NT,TFTF", "Qfff",
+                                        AP_HAL::micros64(),
+                                        (double)max_throttle,
+                                        (double)new_tilt,
+                                        (double)_transition_fwd_tilt_frac);
+
+            } else {
+                new_tilt = deg2quad((float)max_angle_deg);
+            }
+    } else if (quadplane.assisted_flight &&
         transition->transition_state >= Tiltrotor_Transition::TRANSITION_TIMER) {
-        // we are transitioning to fixed wing - tilt the motors all
-        // the way forward
-        slew(get_forward_flight_tilt());
+        // we are transitioning to forward flight on a time based schedule - tilt the motors all
+        // the way forward.
+        // TODO - investigate if the method used for TRANSITION_AIRSPEED_WAIT can be used here.
+        new_tilt = get_forward_flight_tilt();
     } else {
         // until we have completed the transition we limit the tilt to
         // Q_TILT_MAX. Anything above 50% throttle gets
         // Q_TILT_MAX. Below 50% throttle we decrease linearly. This
         // relies heavily on Q_VFWD_GAIN being set appropriately.
        float settilt = constrain_float((SRV_Channels::get_output_scaled(SRV_Channel::k_throttle)-MAX(plane.aparm.throttle_min.get(),0)) * 0.02, 0, 1);
-       slew(MIN(settilt * max_angle_deg * (1/90.0), get_forward_flight_tilt())); 
+       new_tilt = MIN(settilt * deg2quad(max_angle_deg), get_forward_flight_tilt());
     }
+    _transition_fwd_tilt_frac = slew(new_tilt, tilt_rate_dps);
```

Comment:
I think you can remove this variable and use `current_tilt` directly now?

Reply:
The class variable has been replaced by current_tilt as sugggested.